### ISSUE TITLE
CancellationToken issue

### ISSUE THREAD

**Issue by tdjastrzebski** [domaindrivendev/Swashbuckle#744]
Method with signature
`
[HttpGet]
[Route("authorize")]
public async Task<string> Authorize(string username, string password, *CancellationToken* token)
`
**CancellationToken** type gets included with generated swagger, along with **WaitHandle** and **SafeWaitHandle** while, I guess, they should not.
Especially CancellationToken in swagger confuses https://github.com/Azure/autorest/ client code generator.

**Comment by jkrez** [comment]
Hi tdjastrzebski,

There's a thread about ignoring cancellation tokens over here: https://github.com/domaindrivendev/Swashbuckle/issues/306

**Comment by tdjastrzebski** [comment]
Perhaps I did not explain the issue clearly so let me clarify: **method parameter** of CancellationToken type _gets ignored_ and this is great. However, CancellationToken **type** still and for no reason _is included_ with output swagger. This gets https://github.com/Azure/autorest/ client code generator confused - type conflicts etc.

**Comment by johlrich** [comment]
Didn't try with that tool, but I did use the ISchemaFilter to simply clear out the properties array from them as a workaround. The object types still appear in the list, but all the properties are gone and nothing refers to them. I still think they should be filtered from the raw swagger json long term though.

```cs
internal class ClearThreadingTypesWorkaroundFilter : ISchemaFilter
    {
        public void Apply(Schema schema, SchemaRegistry schemaRegistry, Type type)
        {
            if (type.Equals(typeof(System.Threading.CancellationToken)) 
                || type.Equals(typeof(System.Threading.WaitHandle))
                || type.Equals(typeof(Microsoft.Win32.SafeHandles.SafeWaitHandle)))
            {
                schema.properties.Clear();
            }
        }
    }
```

**Comment by tdjastrzebski** [comment]
Here is the solution that worked: modify SwaggerGenerator.CreateOperation() method as follows:

```
private Operation CreateOperation(ApiDescription apiDesc, SchemaRegistry schemaRegistry)
{
	var parameters = apiDesc.ParameterDescriptions
		.Where(p => p.ParameterDescriptor == null || p.ParameterDescriptor.ParameterType.FullName != "System.Threading.CancellationToken")
		.Select(paramDesc =>
			{
				string location = GetParameterLocation(apiDesc, paramDesc);
				return CreateParameter(location, paramDesc, schemaRegistry);
			})
			.ToList();
...
```

**Comment by tdjastrzebski** [comment]
I maintain a private version of Swashbuckle just to make sure CancellationToken is excluded from operation parameters and schema definitions. I was just contemplating whether IOperationFilter could be used for that purpose. It cannot and here is why.

The problem is in SwaggerGenerator.CreateOperation() method logic. Calling SwaggerGenerator.CreateParameter() does more than what the method name suggests. Inside it calls SchemaRegistry.GetOrRegister() method and registers schema in definitions BEFORE operation filters are applied.

**Comment by domaindrivendev** [comment]
Fixed by 7ab9b842c5ad068dd7115a102bed45559133e0a6. Will be automatically excluded with no need for custom filters

**Comment by gpomykala** [comment]
It seems like the fix was never released
